core/perf_test/PerfTest_CustomReduction.cpp

```diff
@@ -58,8 +58,8 @@ void custom_reduction_test(int N, int R, int num_trials) {
   Scalar max;
 
   int team_size = 32;
-  if (team_size > Kokkos::DefaultExecutionSpace::concurrency())
-    team_size = Kokkos::DefaultExecutionSpace::concurrency();
+  if (team_size > Kokkos::DefaultExecutionSpace().concurrency())
+    team_size = Kokkos::DefaultExecutionSpace().concurrency();
```

Comment:
This raises a question: how confident are we that we can generally promise that constructing a default instance of an execution space is and will remain 'cheap'? It's something that happens in a lot of places, but it's never been clear to me if that's actually a commitment Kokkos makes.

Reply:
I mean that's the reason why we are using a static (`Impl`) execution space object for a default-constructed execution space instance.